### ISSUE TITLE
Non-deterministic Travis test fail on OSX/Python2.7

### ISSUE THREAD

**Issue by randomir** [dwavesystems/dwave-cloud-client#98]
Sometimes and only for OSX + Python2.7 builds, one test fails with:
```
.....sssssss.............................................F....ssssssssssssssssssssss...
======================================================================
FAIL: test_submit_continue_then_ok_and_error_reply (tests.test_mock_submission.MockSubmission)
Handle polling for the status of multiple problems.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/travis/build/dwavesystems/dwave-cloud-client/tests/test_mock_submission.py", line 280, in test_submit_continue_then_ok_and_error_reply
    self._check(results1, linear, quad, 100)
AssertionError: SolverFailureError not raised
----------------------------------------------------------------------
Ran 87 tests in 0.413s
FAILED (failures=1, skipped=29)
```

**Comment by randomir** [comment]
Looks like this is still not fixed.

Probably caused by some race condition, or improper limiting of number of poll threads. Probably the complete test should be rewritten.

**Comment by randomir** [comment]
This should be fixed as of #142, more precisely ef29af9904737071c29f4b04d1a33ce94090ed30. The complete test is rewritten and number of both polling workers and **submission workers** is limited to one.
